src/breakupModel/simulation/Collision.cpp

```diff
@@ -61,27 +61,36 @@ void Collision::assignParentProperties() {
 
     //Assign debris the big parent if they are greater than the small parent
     double assignedMassForBigSatellite = 0;
-    for (auto &sat : _output) {
-        if (sat.getCharacteristicLength() > smallSat.getCharacteristicLength()) {
-            sat.setName(debrisNameBigPtr);
-            sat.setVelocity(bigSat.getVelocity());
-            assignedMassForBigSatellite += sat.getMass();
+
+    auto tupleView = _output.getCMVNTuple();
+    //Not parallel! We do not want any race conditions on assignedMassForBigSatellite
```

Comment:
you can make it parallel:
* Option A: make `assignedMassForBigSatellite` a std::atomic and use `fetch_add()` with `std::memory_order_relaxed`. However, this might be slow since there is not much work besides the addition.
* Option B: make two "loops": One that only transforms the data into a vector of numbers, which can be fully parallel, and then apply a `std::reduce` on it, which can also execute parallel.

Reply:
Option A is only possible with C++20 😢 because C++17 does not contain atomic add operations for floating points.
Option B: I'm a bit unsure how to realize that. I will think about it